### ISSUE TITLE
Don't coerce argument dependencies into strings OR change the method definition of all the adapter methods

### ISSUE THREAD

**Issue by ericfranz** [OSC/ood_job#4]
From a discussion on #3.

Here we have: https://github.com/OSC/ood_job/blob/ad8ded0c070157fb7121b557382d262c52c3480a/lib/ood_job/adapters/torque.rb#L45-L48

```ruby
        after      = [after].flatten.map(&:to_s)
        afterok    = [afterok].flatten.map(&:to_s)
        afternotok = [afternotok].flatten.map(&:to_s)
        afterany   = [afterany].flatten.map(&:to_s)
```

The calling `.map(&:to_s)` is understandably proactive but in this case is wrong. The comments above Adapter#submit clearly state that the only acceptable arguments are a String or an array of strings, not any object that implements `to_s`.

Since in the rest of the adapter methods dealing with job ids, we accept only Strings as arguments and do nothing to coerce these, we shouldn't coerce here to Strings.

If we are going to coerce here to Strings, then we should be consistent and coerce to string in all the other Adapter methods that accept id as an argument and then in the comment update it so instead of saying it accepts a "String" saying that it accepts (`#to_s`) or Array of `#to_s`.

But it might be better to just accept Strings:

```ruby
        after      = [after].flatten
        afterok    = [afterok].flatten
        afternotok = [afternotok].flatten
        afterany   = [afterany].flatten
```

Whatever we do though it should be consistent. The current implementation would actually let me write clever code that uses an object or a symbol in place of the String id that submit returned.

My original comment:
>Actually u think it's superfluous and actually incorrect here to do
>
>    after      = [after].flatten.map(&:to_s)
>
>Instead of just
>
>    after      = [after].flatten
>
>The method signature says the adapter needs to support only String or an array of Strings, which are ids returned by the submit method. Not another object that by calling to_s on it would return corresponding job id.
>
>Also this would actually be inconsistent because the other adapter methods  claim they only accept a string for job id as an argument, and we do not coerce the passed in object to a string.
>
>While it would be clever to be able to define #to_s on a Job object to return an id and then be able to do something like adapter.status(id: jobobj) I think then it's still better to do adapter.status(id: jobobj.id) and in the same way instead of adapter.submit(script: postscript, after: jobobj) to do require adapter.submit(script: postscript, after: jobobj.id)
>
>But in most cases it seems that we would never have this scenario. Instead, we submit a job, get a return value, and pass that argument right back to submit as a dependency. And that will always be a string or array of strings.
>

**Comment by nickjer** [comment]
Sure. I can update the docs to specify `#to_s` and call that method for the `id` in the respective methods.

**Comment by ericfranz** [comment]
Sounds good

**Comment by ericfranz** [comment]
>I can just change the Ruby docs to say it accepts #to_s instead of String. I only did it String before because I wanted to make it easy on the developers, but since we are doing the array conversion anyways, then that argument is moot.

I am definitely in favor of making it easy on developers. I just wanted to make sure that if we were going to call `to_s` on `submit` in the TorqueAdapter for dependency ids that:

1. we did the same in the other methods
2. we ensured that future adapters would do the same too

Small chance that we write code that works with one adapter and not with another if it didn't do an expected coercion.
